### PR TITLE
Closes #89: Add SSH connection test

### LINKED ISSUE #89: Add a SSH Connection test
**Is your feature request related to a problem? Please describe.**
If we don't get a proper connection to the server via SSH, test dependent on this could fail for no clear reason, some times a false positive.

**Describe the solution you'd like**
We should test for a clear connection first, log any connection error and fail the test quickly before it even starts.

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by jeawhanlee**
# Description
Add a SSH Connection test and make test fail prematurely if no ssh connection established.

Fixes #89

### CODE REVIEW

**Review by MathieuLamiot — CHANGES_REQUESTED** [review]
Thank you for the PR @jeawhanlee!
I have an issue with it: the ssh command hangs forever, blocking the execution of the test plan. It seems to be because we just call `ssh ubuntu@xx.xx.xx.xx` and this command does not return, it just opens an ssh terminal and waits.

I validated my thoughts by changing the code to:

```
export async function testSshConnection(): Promise<string> {
    if(configurations.type !== ServerType.external) {
        return;
    }
    console.log("Hello");
    const cwd = getWPDir(configurations);
    console.log("Ready?");
    const command: string = `ssh ${configurations.ssh.username}@${configurations.ssh.address} -i ${configurations.ssh.key}`;
    
    const result = exec(command);
    console.log("Done");
```

And giving me the output:

```
npm run test:lcp

> wp-rocket-e2e@1.0.0 test:lcp
> $npm_package_config_testCommand --tags @lcp

This Node.js version (v21.6.2) has not been tested with this version of Cucumber; it should work normally, but please raise an issue if you see anything unexpected.
Hello
Ready?
Pseudo-terminal will not be allocated because stdin is not a terminal.
Welcome to Ubuntu 23.04 (GNU/Linux 6.2.0-39-generic x86_64)

 * Documentation:  https://help.ubuntu.com
 * Management:     https://landscape.canonical.com
 * Support:        https://ubuntu.com/pro

  System information as of Thu Jun  6 15:03:44 UTC 2024

  System load:           1.15
  Usage of /:            38.1% of 48.30GB
  Memory usage:          41%
  Swap usage:            0%
  Processes:             120
  Users logged in:       1
  IPv4 address for ens3: 51.91.147.85
  IPv6 address for ens3: 2001:41d0:404:100::24e8


0 updates can be applied immediately.


The list of available updates is more than a week old.
To check for new updates run: sudo apt update
Your Ubuntu release is not supported anymore.
For upgrade information, please visit:
http://www.ubuntu.com/releaseendoflife

New release '23.10' available.
Run 'do-release-upgrade' to upgrade to it.

```

I never get to the 'Done' log.

Then, I changed the command to:

const command: string = `ssh ${configurations.ssh.username}@${configurations.ssh.address} -i ${configurations.ssh.key} env`;
and now it works: the code continues correctly.

Also I checked by putting a wrong username, and I get the error raised.

WDYT of this change?

File: utils/commands.ts [hunk 1]
```diff
@@ -395,4 +395,21 @@ export async function getWPTablePrefix(): Promise<string> {
     return tablePrefix;
 }
 
+export async function testSshConnection(): Promise<string> {
+    if(configurations.type !== ServerType.external) {
+        return;
+    }
+
+    const cwd = getWPDir(configurations);
```
Comment (MathieuLamiot) [inline utils/commands.ts]: This is not used?

**Review by MathieuLamiot — CHANGES_REQUESTED** [review]

File: utils/commands.ts (hunk 1 above)
Comment (jeawhanlee) [reply utils/commands.ts]: Ah Yes

Comment (jeawhanlee) [reply utils/commands.ts]: Good Catch

**Review by MathieuLamiot — APPROVED** [review]

**Comment by MathieuLamiot** [comment]
tested:
- with just the PR: all OK
- changing the ssh user, I get the error thrown.

**Pull request merged**